com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/resource/IAdditionalExport.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+
+package com.avaloq.tools.ddk.xtext.resource;
+
+import java.util.Set;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.naming.QualifiedName;
+import org.eclipse.xtext.resource.IEObjectDescription;
+import org.eclipse.xtext.util.IAcceptor;
+
+
+/**
+ * Extensions for export.
+ */
+public interface IAdditionalExport {

```

Comment:
I noticed that almost none of the methods in here are called by the DDK itself. Rather than only providing the plumbing I think it would make sense if the implementations of the corresponding DDK services were to already include the support for these extensions, so that the DSLs don't have to "hack" this. This was probably the idea to begin with, but I wonder if we shouldn't do this from the beginning.

Reply:
Yep, just wanted to start with one language. Will do a bit more. I am extending the generator so at least the extension itself does not need to do the manual plumbing. The languages that allow extensions are not so many, so I would anyway prefer to have an "opt in" option for the languages to support the extensions. Sure, most of the code will have to go up to the DDK, but the switch will be in the language. Will try to balance between having a cleaner solution upfront and not blocking Ruben for too much who is waiting for this mechanism.